homeassistant/components/reolink/number.py

```diff
@@ -62,10 +63,103 @@ class ReolinkNumberEntityDescription(
         native_step=1,
         get_min_value=lambda api, ch: api.zoom_range(ch)["focus"]["pos"]["min"],
         get_max_value=lambda api, ch: api.zoom_range(ch)["focus"]["pos"]["max"],
-        supported=lambda api, ch: api.zoom_supported(ch),
+        supported=lambda api, ch: api.supported(ch, "zoom"),
         value=lambda api, ch: api.get_focus(ch),
         method=lambda api, ch, value: api.set_zoom(ch, int(value)),
     ),
+    ReolinkNumberEntityDescription(
+        key="floodlight_brightness",
+        name="Floodlight brightness",
```

Comment:
Looks like this is handled by the `light` entity in https://github.com/home-assistant/core/pull/88619/files#diff-0248158d2443bae1c91d1ec20d1335a8a86a69b421c2a258f843a71c1027c957R45-R56

Reply:
No not true, when the floodlight mode is set to "Auto" or "Schedule", the camera will automatically turn on/off the floodlight internally independent of the light entity. The Brightness of this number entity applies to both this automatic turn on of the floodlight and when manually turned on through the light entity.

The light entity will only allow you to change the brightness when also actually turning on the floodlight, while this number entity allows you to change the brightness while the spotlight keeps off/is controlled internally by the camera floodlight mode ("Auto" or "Scheduale")

see: 
https://github.com/home-assistant/core/pull/87946/files#diff-24a4e9eab3c1a3ca00959be0cca50b3701b27598b6ea20bf268b3a5cef5004d8R39-R48

Reply:
I have to agree partly with @cgarwood, this is highly confusing towards an end user.
I had to read the above 2 times, to even remotely understand what is going on. A user won't be able to figure this out by looking at the Home Assistant UI and their entities.

That latter indicates something is wrong IMHO.

Reply:
@frenck I know, it is also a bit confusing in the reolink app/client.
Would you have any suggestions on how I could make this more clear for the end user?

Maybe rename to "Global floodlight brightness"?
And/or remove the brightness control from the light.floodlight entity so this is the only floodlight brightness in HomeAssistant?

Reply:
yeah or maybe "Floodlight turn on brightness"?

Reply:
Good suggestion, I renamed to "Floodlight turn on brightness".
Schould I remove the brightness control from the light.floodlight entity or just leave it?
(It will do the same thing as this number entity, but since you have to use the light.turn_on service it will also turn the floodlight manually on untill you turn it off again using light.turn_off).

Reply:
> Schould I remove the brightness control from the light.floodlight entity or just leave it?

I think that one can be left as is.